docs/source/howto/codes/interfacing.rst

```diff
@@ -0,0 +1,282 @@
+.. _how-to:codes:plugin:
+
+Interfacing external codes
+==========================
+
+In order to run an external code, you will first have to explain to AiiDA how to run it: how to create inputs files from the content of the input node, how the program can read those input files, where to find and how to read the output files, etc.
+The way you transmit these instructions to AiiDA is by extending the |CalcJob| and the |Parser| classes:
+
+.. code:: python
+
+    # -*- coding: utf-8 -*-
+    from aiida.engine import CalcJob
+
+    class ArithmeticAddCalculation(CalcJob):
+
+        @classmethod
+        def define(cls, spec):
+            super().define(spec)
+            # Specifications of inputs, outputs and exit codes.
+            # no return statement
+
+        def prepare_for_submission(self, folder):
+            # Instructions on how to prepare the input files
+            return calcinfo
+
+.. code:: python
+
+    # -*- coding: utf-8 -*-
+    from aiida.parsers.parser import Parser
+
+    class ArithmeticAddParser(Parser):
+
+        def parse(self, **kwargs):
+            # Instructions on how to process the output files
+            # no return statement
+
+As you can guess from the name of the child-classes in the previous code snippet, we will illustrate the implementation of this code wrapping procedure with a possible design for the example introduced in the :ref:`tutorial<tutorial:basic>` of a code that sums two numbers.
+To have a clearer idea of how to interact with this code, we can assume it consists of the following bash script which basically reads two consecutive integers in a single line of an input file provided through a command line argument and write to standard output the result of its sum (so one would run it as ``$ add.sh -in [input_file] > [output_file]``).
+
+.. code:: bash
```

Comment:
Consider placing the code for the `add.sh` script in  dropdown or possibly don't provide it at all. The how-to section is not a tutorial, so you can use examples, but there is no expectation that users can follow them step by step.

Reply:
Here it might be better to go with the changes of Marnik, which does not require a separate script at all, but just uses `bash`. Since this can typically be found always at the same place, it makes it very easy to configure the code (and even provide a complete configuration yaml file) and easier to explain, as we are not executing some custom script, but just bash.